### ISSUE TITLE
Question about ‘labels’

### ISSUE THREAD

**Issue by SatireY** [meta-llama/llama-recipes#382]
Hi, thanks for your great work!
I'm using the code in the llama-recipes repository to fine-tune my own data and have a question for you: I read the examples/quickstart.ipynb code, I found that when preparing data, not only need to [input_ids, attention_mask] two columns of data, also need to 'labels' column data, I would like to know what role the ‘labels’ column data? Can I provide only [input_ids, attention_mask] two columns for training?
Looking forward to your reply!

**Comment by wukaixingxp** [comment]
Hi! Please take a look at [this example](https://github.com/meta-llama/llama-recipes/blob/main/src/llama_recipes/datasets/samsum_dataset.py#L32), where we masked out the prompt label to [-100] to avoid computing loss on the prompts. However, in Llama 3, we recommend to use first use [tokenizer.apply_chat_template](https://github.com/meta-llama/llama-recipes/blob/main/recipes/finetuning/datasets/custom_dataset.py#L15), then [create labels for prompts](https://github.com/meta-llama/llama-recipes/blob/main/recipes/finetuning/datasets/custom_dataset.py#L24) as shown in the custom_dataset.py

**Comment by HamidShojanazeri** [comment]
@SatireY the idea here to provide labels some how that you can calculate the loss, ideally your mask token/ next token is your label but if you are fine-tuning on a instruction tuning job, you wan to calculate loss only on generation not context, that what we provide in our dataset implementations.
